File: Basic Programming/Dimik oj/C++/প্রোগ্রামিং সমস্যা 41 - [৫২ সমস্যা বই] ধারার যোগফল-২.cpp
/*                    Name : Arif Khan
                      Judge: Dimik Oj
                      University: Primeasia University
                      problem: প্রোগ্রামিং সমস্যা 41 - [৫২ সমস্যা বই] ধারার যোগফল-২
                      Difficulty: Easy
                      Time Complexity: 0.2683 সেকেন্ড
                      Problem Link: https://dimikoj.com/problems/41/sum-of-the-series-2-by
*/

#include <bits/stdc++.h>
using namespace std;
#define ll long long int

ll fact(ll n){
    ll i , p = 1;
    for(i = 1; i <= n; i++) p *= i;
    return p;
}

int main() {
    int n;
	cin >> n;
	while(n--) {
	    double sum = 0.0;
	    ll x;
	    cin >> x;
	    for(ll i = 1; i <= x; i++){
	        sum += (double(i)/fact(i));
	    }
	    printf("%.4lf\n",sum);
	}
	return 0;
}

File: Basic Programming/Dimik oj/Python3/প্রোগ্রামিং সমস্যা 41 - [৫২ সমস্যা বই] ধারার যোগফল-২.py
#                      Name : Arif Khan
#                      Judge: Dimik Oj
#                      University: Primeasia University
#                      problem: প্রোগ্রামিং সমস্যা 41 - [৫২ সমস্যা বই] ধারার যোগফল-২
#                      Difficulty: Easy
#                      Time Complexity: 0.6055 সেকেন্ড
#                      Problem Link: https://dimikoj.com/problems/41/sum-of-the-series-2-by
#

def fac(n):
    p = 1
    for i in range(1,n+1):
        p *= i
    return p
    
for _ in range(int(input())):
    x = int(input())
    sum = 0.0
    for i in range(1,x+1):
        sum += (float(i)/fac(i))
    print("%.4f"%(sum))